File: backend/app/models/wishlist.py
"""
Wishlist model
"""
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, Date, Enum, ForeignKey
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base
import enum


class WishlistTypeEnum(str, enum.Enum):
    """Wishlist type enum"""
    PERMANENT = "permanent"
    EVENT = "event"


class VisibilityEnum(str, enum.Enum):
    """Visibility enum"""
    PUBLIC = "public"
    BY_LINK = "by_link"
    FRIENDS_ONLY = "friends_only"


class Wishlist(Base):
    """Wishlist model"""
    __tablename__ = "wishlists"
    
    id = Column(Integer, primary_key=True, index=True)
    owner_id = Column(Integer, ForeignKey("users.id", ondelete="CASCADE"), nullable=False, index=True)
    title = Column(String(255), nullable=False)
    description = Column(Text, nullable=True)
    cover_image_url = Column(String(500), nullable=True)
    cover_emoji = Column(String(10), nullable=True)  # For emoji covers
    
    # Тип вишлиста
    wishlist_type = Column(Enum(WishlistTypeEnum), default=WishlistTypeEnum.PERMANENT, nullable=False)
    event_name = Column(String(255), nullable=True)
    event_date = Column(Date, nullable=True)
    
    # Приватность
    visibility = Column(Enum(VisibilityEnum), default=VisibilityEnum.BY_LINK, nullable=False)
    share_token = Column(String(64), unique=True, nullable=False, index=True)
    
    is_archived = Column(Boolean, default=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    # Relationships
    owner = relationship("User", back_populates="wishlists")
    items = relationship("Item", back_populates="wishlist", cascade="all, delete-orphan")

File: wishliste/wishliste/Domain/Models/Wishlist.swift
import Foundation

struct Wishlist: Codable, Equatable {
    let id: Int
    let ownerId: Int
    var title: String
    var description: String?
    var coverImageUrl: String?
    var coverEmoji: String?
    var wishlistType: WishlistType
    var eventName: String?
    var eventDate: String?
    var visibility: Visibility
    let shareToken: String
    var isArchived: Bool
    let createdAt: String
    var updatedAt: String?
    var items: [Item]?
    var itemsCount: Int?

    enum CodingKeys: String, CodingKey {
        case id, title, description, visibility
        case ownerId = "owner_id"
        case coverImageUrl = "cover_image_url"
        case coverEmoji = "cover_emoji"
        case wishlistType = "wishlist_type"
        case eventName = "event_name"
        case eventDate = "event_date"
        case shareToken = "share_token"
        case isArchived = "is_archived"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case items
        case itemsCount = "items_count"
    }
}

enum WishlistType: String, Codable, CaseIterable {
    case permanent
    case event
}

enum Visibility: String, Codable, CaseIterable {
    case `public`
    case byLink = "by_link"
    case friendsOnly = "friends_only"
}
